airbyte-integrations/connectors/destination-azure-blob-storage/src/main/java/io/airbyte/integrations/destination/azure_blob_storage/AzureBlobStorageConsumer.java

```diff
@@ -84,39 +94,30 @@ protected void startTracked() throws Exception {
     }
   }
 
-  private boolean createContainers(final AppendBlobClient appendBlobClient,
-                                   final ConfiguredAirbyteStream configuredStream) {
+  private void createContainers(final SpecializedBlobClientBuilder specializedBlobClientBuilder,
+                                final AppendBlobClient appendBlobClient,
+                                final ConfiguredAirbyteStream configuredStream) {
     // create container if absent (aka SQl Schema)
     final BlobContainerClient containerClient = appendBlobClient.getContainerClient();
     if (!containerClient.exists()) {
       containerClient.create();
     }
-    // create a storage container if absent (aka Table is SQL BD)
-    if (SyncMode.FULL_REFRESH.equals(configuredStream.getSyncMode())) {
-      // full refresh sync. Create blob and override if any
+    if (DestinationSyncMode.OVERWRITE.equals(configuredStream.getDestinationSyncMode())) {
       LOGGER.info("Sync mode is selected to OVERRIDE mode. New container will be automatically"
-          + " created or all data would be overridden (if any) for stream:" + configuredStream
+              + " created or all data would be overridden (if any) for stream:" + configuredStream
               .getStream().getName());
-      appendBlobClient.create(true);
-      return true;
-    } else {
-      // incremental sync. Create new container only if still absent
-      if (!appendBlobClient.exists()) {
-        LOGGER.info("Sync mode is selected to APPEND mode. New container will be automatically"
-            + " created for stream:" + configuredStream.getStream().getName());
-        appendBlobClient.create(false);
-        LOGGER.info(appendBlobClient.getBlobName() + " blob has been created");
-        return true;
-      } else {
-        LOGGER.info(String.format(
-            "Sync mode is selected to APPEND mode. Container %s already exists. Append mode is "
-                + "only available for \"Append blobs\". For more details please visit"
-                + " https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#blobs",
-            configuredStream.getStream().getName()));
-        LOGGER.info(appendBlobClient.getBlobName() + " already exists");
-        return false;
-      }
+      var blobItemList = StreamSupport.stream(containerClient.listBlobs().spliterator(), false)
+              .collect(Collectors.toList());
+      blobItemList.forEach(blob -> {
+        if (!blob.isDeleted() && blob.getName().contains(configuredStream.getStream().getName() + "/")) {
```

Comment:
Could `blob.getName()` throw NullPointer here? The same question for `configuredStream.getStream()` ?

Reply:
A blob always has a name, just like a stream for configuredStream. Therefore appearance NPE is impossible.